cvat-ui/src/components/annotation-page/annotations-actions/annotations-actions-modal.tsx

```diff
@@ -0,0 +1,584 @@
+// Copyright (C) 2023 CVAT.ai Corporation
+//
+// SPDX-License-Identifier: MIT
+
+import './styles.scss';
+
+import React, {
+    useEffect, useReducer, useRef, useState,
+} from 'react';
+import Button from 'antd/lib/button';
+import { Col, Row } from 'antd/lib/grid';
+import Progress from 'antd/lib/progress';
+import Select from 'antd/lib/select';
+import notification from 'antd/lib/notification';
+import Text from 'antd/lib/typography/Text';
+import Modal from 'antd/lib/modal';
+
+import config from 'config';
+import { useIsMounted } from 'utils/hooks';
+import { createAction, ActionUnion } from 'utils/redux';
+import { getCVATStore } from 'cvat-store';
+import { BaseSingleFrameAction, Job, getCore } from 'cvat-core-wrapper';
+import { Canvas } from 'cvat-canvas-wrapper';
+import { fetchAnnotationsAsync, saveAnnotationsAsync } from 'actions/annotation-actions';
+import { switchAutoSave } from 'actions/settings-actions';
+import { Alert, InputNumber } from 'antd';
+import { clamp } from 'utils/math';
+
+const core = getCore();
+
+interface State {
```

Comment:
Why all actions state code is here? Isnt it contradicting with our approach with all in one state?

Reply:
1. I do not think we have such approach
2. Actions state here because other app does not need to know about actions state. It is encapsulated. 
3. Finally this component is even not in context of redux state because it was rendered with ReactDOM. Literally it is another react mini application (like typical static Modal (Modal.confirm(), etc, for example)